workers/common/src/main/kotlin/common/OrtRunService.kt

```diff
@@ -489,4 +499,32 @@ class OrtRunService(
             ortRunRepository.update(ortRunId, issues = issues.asPresent())
         }
     }
+
+    /**
+     * For scan results regarding a [RepositoryProvenance], if this provenance has a VCS path, then filter
+     * out the [OrtScanResult] summaries that are not related to this path. These summaries typically contain
+     * findings about licenses, copyrights, code snippets and other information from the scan.
+     */
+    private fun filterScanResultSummary(
+        provenances: Collection<ProvenanceResolutionResult>,
+        scanResults: Set<ScanResult>
+    ): Set<OrtScanResult> {
+        val vcsPaths = provenances
+            ?.mapNotNull { (it.packageProvenance as? RepositoryProvenance)?.vcsInfo?.path }
```

Comment:
`provenances` is not nullable, so the Elvis-operators are not needed here, allowing to simplify code. For readability maybe also consider filtering for `RepositoryProvenance` instances separately, but together with the mapping that would be two times iterating `provenances`, so maybe not desirable.